### DISCUSSION TITLE
defaultMaxListeners exceed in budibase app

### DISCUSSION THREAD

**Discussion by dpaes** [Budibase/budibase#14746] [Help] [answered]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
- Self
  - Method: docker compose
  - Budibase Version: latest
  - App Version: latest

**Describe the bug**
i'm trying to use budibase with docker compose yml file shared in docker installation selfhosted but the budibase app fails with the logs shared here. socket errors, something like emitter events with defaultMaxListeners exceed.

**To Reproduce**
only install using docker compose file

**Expected behavior**
work like roadmap

**Screenshots**
![image](https://github.com/user-attachments/assets/bf1131a6-a757-4768-ad77-da4231202544)
I'm trying to use budibase latest, using the docker roadmap but with redis and minio already installed, then using docker compose to deploy budibase app, worker and couchDB in it. 

**App Export**
If possible - please attach an export of your budibase application for debugging/reproduction purposes.
ERROR timestamp=2024-10-07T17:23:56.793Z service=@budibase/server pid=29 tenantId=default msg=Failed server startup - request to https://budibase-back.credgs.com.br/api/global/configs/checklist failed, reason: self-signed certificate

ERROR timestamp=2024-10-07T17:23:56.793Z service=@budibase/server err={"type":"FetchError","message":"request to https://budibase-back.credgs.com.br/api/global/configs/checklist failed, reason: self-signed certificate","stack":"FetchError: request to https://budibase-back.credgs.com.br/api/global/configs/checklist failed, reason: self-signed certificate\n    at ClientRequest.<anonymous> (/app/node_modules/node-fetch/lib/index.js:1491:11)\n    at ClientRequest.emit (node:events:519:28)\n    at ClientRequest.emit (node:domain:488:12)\n    at emitErrorEvent (node:_http_client:108:11)\n    at TLSSocket.socketErrorListener (node:_http_client:511:5)\n    at TLSSocket.emit (node:events:519:28)\n    at TLSSocket.emit (node:domain:488:12)\n    at emitErrorNT (node:internal/streams/destroy:169:8)\n    at emitErrorCloseNT (node:internal/streams/destroy:128:3)\n    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)","errno":"DEPTH_ZERO_SELF_SIGNED_CERT","code":"DEPTH_ZERO_SELF_SIGNED_CERT","name":"FetchError"} pid=29 tenantId=default msg=bb-alert: Uncaught exception.

INFO timestamp=2024-10-07T17:23:56.794Z service=@budibase/server pid=29 tenantId=default msg=Server Closed

INFO timestamp=2024-10-07T17:23:59.824Z service=@budibase/server pid=29 tenantId=default msg=Queues shutdown

INFO timestamp=2024-10-07T17:23:59.825Z service=@budibase/server pid=29 tenantId=default msg=REDIS SHUTDOWN

INFO timestamp=2024-10-07T17:23:59.829Z service=@budibase/server pid=29 tenantId=default msg=Redis shutdown

INFO timestamp=2024-10-07T17:23:59.829Z service=@budibase/server pid=29 tenantId=default msg=Events shutdown

ERROR timestamp=2024-10-07T17:23:59.831Z service=@budibase/server pid=29 tenantId=default msg=(node:29) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [Socket]. MaxListeners is 10. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)

INFO timestamp=2024-10-07T17:23:59.831Z service=@budibase/server pid=29 tenantId=default msg=Threads shutdown

error Command failed with exit code 255.

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version latest

**Additional context**

**Comment by linear** [comment]
<p><a href="https://linear.app/budibase/issue/BUDI-8701/defaultmaxlisteners-exceed-in-budibase-app">BUDI-8701 defaultMaxListeners exceed in budibase app</a></p>

**Comment by mike12345567** [comment]
Hi @dpaes - have you changed your `docker-compose.yaml` at all from the default found here: https://github.com/Budibase/budibase/blob/master/hosting/docker-compose.yaml

Please confirm if an installation using this default setup works for you.

**Reply by dpaes** [reply]
yes i have changed to use what in compose file i already have, like minio, proxy and redis, so with that change, i only use the budibase-app, worker and couchDB service

**Reply by mike12345567** [reply, marked as answer]
Hi @dpaes - can you attempt to run the service with the default docker-compose file? It is difficult for us to support all custom configurations so generally we ask to confirm the services all work as expected when using MinIO, Redis and Proxy as per the default setup.
